cabal-install/main/Main.hs

```diff
@@ -447,40 +448,67 @@ reconfigureAction flags@(configFlags, _) _ globalFlags = do
     checkFlags [] globalFlags config
   pure ()
 
-buildAction :: (BuildFlags, BuildExFlags) -> [String] -> Action
-buildAction (buildFlags, buildExFlags) extraArgs globalFlags = do
-  let verbosity = fromFlagOrDefault normal (buildVerbosity buildFlags)
-      noAddSource = fromFlagOrDefault DontSkipAddSourceDepsCheck
-                    (buildOnly buildExFlags)
-  (useSandbox, config) <- loadConfigOrSandboxConfig verbosity globalFlags
-  distPref <- findSavedDistPref config (buildDistPref buildFlags)
-  -- Calls 'configureAction' to do the real work, so nothing special has to be
-  -- done to support sandboxes.
-  config' <-
-    reconfigure configureAction
-    verbosity distPref useSandbox noAddSource (buildNumJobs buildFlags)
-    mempty [] globalFlags config
-  nixShell verbosity distPref globalFlags config $ do
-    maybeWithSandboxDirOnSearchPath useSandbox $
-      build verbosity config' distPref buildFlags extraArgs
 
+buildAction :: (BuildFlags, BuildExFlags) -> [String] -> Action
+buildAction flags@(buildFlags, _) = buildActionForCommand
+  (Cabal.buildCommand defaultProgramDb)
+  verbosity
+  flags
+  where verbosity = fromFlagOrDefault normal (buildVerbosity buildFlags)
+
+-- showBuildInfoAction :: (Cabal.ShowBuildInfoFlags, BuildExFlags) -> [String] -> Action
+-- showBuildInfoAction (showBuildInfoFlags, buildEx) = buildActionForCommand
+--   (Cabal.showBuildInfoCommand defaultProgramDb)
+--   showBuildInfoFlags
+--   verbosity
+--   (Cabal.buildInfoBuildFlags showBuildInfoFlags, buildEx)
+--     -- Default silent verbosity so as not to pollute json output
+--   where verbosity = fromFlagOrDefault silent (buildVerbosity (Cabal.buildInfoBuildFlags showBuildInfoFlags ))
```

Comment:
I think you moved this to a different file so you can probably just remove this?

Reply:
That also seems to make the whole buildActionForCommand refactoring unused, can you have another look at what's going on here?